lib/gettext_i18n_rails/ruby_gettext_extractor.rb

```diff
@@ -127,13 +122,22 @@ def new_call recv, meth, args = nil
     end
   end
 
-  class Extractor18 < Ruby18Parser
-    include ExtractorMethods
+
+  def get_ruby_parser
+    if RUBY_VERSION =~ /^1\.8/
+      Ruby18Parser
+    elsif RUBY_VERSION =~ /^1\.9/
+      Ruby19Parser
+    elsif RUBY_VERSION =~ /^2\.0/
+      Ruby20Parser
+    elsif RUBY_VERSION =~ /^2\.1/
+      Ruby21Parser
+    else
+      Ruby22Parser
+    end
   end
 
-  class Extractor19 < Ruby19Parser
+  class Extractor < get_ruby_parser
```

Comment:
... inheriting from a method :D 
can you assign a constant like `BaseParser` or so instead ?